### PR TITLE
Add support for Jenkins Pipeline

### PR DESCRIPTION
**Pull request by DDRBoxman**
Cleaned up the commit from @ansoni in hope of getting this merged.

### CODE REVIEW

**Comment by nwalke** [comment]
@suryanarayanan Can we get this merged soon?  The pipeline support PRs have been sitting around for quite a while.

**Comment by p1c2u** [comment]
+1

File: src/main/java/com/amazonaws/codedeploy/AWSCodeDeployPublisher.java [hunk 1]
```diff
@@ -31,21 +31,16 @@
...
 
-import hudson.FilePath;
-import hudson.Launcher;
-import hudson.Extension;
-import hudson.Util;
-import hudson.model.AbstractBuild;
-import hudson.model.BuildListener;
-import hudson.model.AbstractProject;
-import hudson.model.Result;
+import hudson.*;
```
Comment (naveedahmedsiddiqui) [inline src/main/java/com/amazonaws/codedeploy/AWSCodeDeployPublisher.java]: We should explicitly import the class we need instead.

**Comment by DDRBoxman** [comment]
This code is released under Apache 2.0

**Comment by cfoehrdes** [comment]
Can we provide any help to get this feature merged in the near future?

**Comment by DDRBoxman** [comment]
Apparently, the team in charge of this changed recently. Amazon said that it might be at least another week till someone looks at it.

**Comment by Jmcfar** [comment]
Hi!

Thanks for the contribution.

Did you perform any testing as part of the changes? If so it'd be helpful to include that in the PR.

**Comment by Jmcfar** [comment]
Manually did testing on our end for both new pipelines support and older style behavior so going to merge this in. Thanks for the contribution!

**Comment by Jmcfar** [comment]
This has been published to the Jenkins plugin repository. Updating to the CodeDeploy plugin 1.19 or later should enable Pipeline support.

**Pull request merged**